MiniSim/Model/Platform.swift

```diff
@@ -8,8 +8,8 @@
 import Foundation
 
 enum Platform: String, Codable {
-    case ios = "ios"
-    case android = "android"
+    case ios
```

Comment:
I remember there were some issues when there wasn't any string value. But maybe before there was no `enum Platform: String`. What happens if we don't provide any values?

Reply:
By default, cases have string raw value, even if you omit `String` type of the `enum`.
![Screenshot 2023-11-10 at 20 14 32](https://github.com/okwasniewski/MiniSim/assets/12799/2610cb7b-4c5e-404c-8e64-4fea2706e44f)

Reply:
The difference is subtle yet really important when you are persisting JSON encoded data. The encoding output is different, which makes it not compatible with older versions. We need to keep the explicit strings

Here is the code I've used: 

```swift
enum PlatformString: String, Codable {
    case ios = "ios"
    case android = "android"
}

enum PlatformNoString: Codable {
    case ios
    case android
}

struct TestStruct: Codable {
    var title: String
    var platform: PlatformString
}

struct TestStructNoString: Codable {
    var title: String
    var platform: PlatformNoString
}
// ... 

let test1 = TestStruct(title: "String", platform: .android)
        let test2 = TestStructNoString(title: "No string", platform: .android)
        if let encoded = try? JSONEncoder().encode(test1) {
            print(String(data: encoded, encoding: .utf8))
        }
        if let encoded = try? JSONEncoder().encode(test2) {
            print(String(data: encoded, encoding: .utf8))
        }

```

And result: 

![CleanShot 2023-11-12 at 15 11 09@2x](https://github.com/okwasniewski/MiniSim/assets/52801365/711e14b1-0899-4c49-8491-168a43647d3a)

Reply:
Actually it depends on the type of the enum. 🙂 If you change `PlatformNoString` to be `String` type, it will encode correctly.

Reply:
True, I didn't check properly